### PR TITLE
WSSQL-158 Remove support for centos5/ubuntu12 due to EoL

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
Removed the following dependency files:
centos 5         (el5.cmake.in)
ubuntu 12.04 (precise.cmake.in)
ubuntu 12.10 (quantal cmake.in)

Added a cmake WARNING to the message output if developer is building on a platform we don't recognize.

**Comment by Michael-Gardner** [comment]
@rpastrana Cleaned out all the old unsupported version files.  I put el5.cmake.in back in, because Ming suggested that we retain the ability to support bug fixes for a few more months for that platform.  Just in case customers haven't moved to a more recent distro.

File: CMakeLists.txt [hunk 1]
```diff
@@ -282,7 +282,7 @@ IF ( NOT MAKE_DOCS_ONLY )
         CONFIGURE_FILE(${HPCC_WSSQL_SOURCE_DIR}/cmake_modules/dependencies/${packageRevision}.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/dependencies/${packageRevision}.cmake)
         INCLUDE(${CMAKE_CURRENT_BINARY_DIR}/dependencies/${packageRevision}.cmake)
     ELSE()
-        MESSAGE("-- WARNING: DEPENDENCY FILE FOR ${packageRevision} NOT FOUND, Using deps template.")
```
Comment (rpastrana) [inline CMakeLists.txt]: Don't these message tend to have the -- prefix?

File: cmake_modules/dependencies/yakkety.cmake.in [hunk 2]
```diff
@@ -1,2 +1,3 @@
+#UBUNTU 16.04
```
Comment (rpastrana) [inline cmake_modules/dependencies/yakkety.cmake.in]: looks like we have #UBUNTU 16.04 here and in the zesty file. one should be 17.xx right?

File: CMakeLists.txt (hunk 1 above)
Comment (Michael-Gardner) [inline CMakeLists.txt]: It's going to throw a warning now, which is much more visible.

example:
{code}
CMake Warning at cmake_modules/commonSetup.cmake:589 (message):
  USE_OPTIONAL set - missing dependencies for optional features will
  automatically disable them
Call Stack (most recent call first):
  CMakeLists.txt:142 (include)

{code}

**Comment by Michael-Gardner** [comment]
Fixed the version comment in the yakkety and zesty file.

File: CMakeLists.txt (hunk 1 above)
Comment (rpastrana) [inline CMakeLists.txt]: gotcha.

**Pull request merged**